[LBSE] Cleanup RenderSVGContainer related code

nikolaszimmermann opened the pull request:
#### 0113dff599ec80b55049327eee2e378abd1da1c8
<pre>
[LBSE] Cleanup RenderSVGContainer related code
<a href="https://bugs.webkit.org/show_bug.cgi?id=242718">https://bugs.webkit.org/show_bug.cgi?id=242718</a>

Reviewed by Rob Buis.

This unifies the main pieces of the container child layout logic in
RenderSVGContainer::layout(), in such a way that the logic is more
straightforward to follow (not having critical pieces of the logic
sprinkled across RenderSVG(Transformable|Viewport)Container) and to
assure that sensitive flags like &apos;m_didTransformToRootUpdate&apos; and
&apos;m_isLayoutSizeChanged&apos; are computed and stored in one place
instead of one per container renderer.

Use SetScope&lt;&gt; instead of manual updates of these flags to avoid any
undefined states.

The RenderSVGContainer layout logic now looks almost identical to
RenderSVGRoot - the LBSE variants are clean &amp; small compared to their
LegacyRenderSVG* counterparts.

Covered by existing tests, that were broken under LBSE before.
(No progressions as the affected tests are still broken compared
to legacy, due to other reasons, e.g. missing &lt;image&gt; support).

* Source/WebCore/rendering/svg/RenderSVGContainer.cpp:
(WebCore::RenderSVGContainer::layout):
(WebCore::RenderSVGContainer::calculateViewport): Deleted.
* Source/WebCore/rendering/svg/RenderSVGContainer.h:
(WebCore::RenderSVGContainer::isLayoutSizeChanged const):
(WebCore::RenderSVGContainer::didTransformToRootUpdate const):
(WebCore::RenderSVGContainer::updateLayoutSizeIfNeeded):
* Source/WebCore/rendering/svg/RenderSVGTransformableContainer.cpp:
(WebCore::RenderSVGTransformableContainer::graphicsElement const):
(WebCore::RenderSVGTransformableContainer::additionalContainerTranslation const):
(WebCore::RenderSVGTransformableContainer::updateFromStyle):
(WebCore::RenderSVGTransformableContainer::updateLayerTransform):
(WebCore::RenderSVGTransformableContainer::applyTransform const):
(WebCore::RenderSVGTransformableContainer::calculateViewport): Deleted.
(WebCore::RenderSVGTransformableContainer::layoutChildren): Deleted.
(WebCore::RenderSVGTransformableContainer::styleWillChange): Deleted.
* Source/WebCore/rendering/svg/RenderSVGTransformableContainer.h:

Canonical link: <a href="https://commits.webkit.org/252466@main">https://commits.webkit.org/252466@main</a>
</pre>

rwlbuis commented on Source/WebCore/rendering/svg/RenderSVGTransformableContainer.cpp:
```diff
@@ -59,66 +63,39 @@ inline SVGUseElement* associatedUseElement(SVGGraphicsElement& element)
     if (associatedUseElement(graphicsElement())) {
-        setHasSVGTransform();
         setHasTransformRelatedProperty();
+        setHasSVGTransform();
```
is this change needed?

nikolaszimmermann replied:
Consistency, all other call sites use this order (See updateSVGTransformFlags())

rwlbuis reviewed (changes requested):
Note in commit message uny -> any.

rwlbuis reviewed (approved):
LGTM.

nikolaszimmermann committed:
[LBSE] Cleanup RenderSVGContainer related code
https://bugs.webkit.org/show_bug.cgi?id=242718

Reviewed by Rob Buis.

This unifies the main pieces of the container child layout logic in
RenderSVGContainer::layout(), in such a way that the logic is more
straightforward to follow (not having critical pieces of the logic
sprinkled across RenderSVG(Transformable|Viewport)Container) and to
assure that sensitive flags like 'm_didTransformToRootUpdate' and
'm_isLayoutSizeChanged' are computed and stored in one place
instead of one per container renderer.

Use SetScope<> instead of manual updates of these flags to avoid any
undefined states.

The RenderSVGContainer layout logic now looks almost identical to
RenderSVGRoot - the LBSE variants are clean & small compared to their
LegacyRenderSVG* counterparts.

Covered by existing tests, that were broken under LBSE before.
(No progressions as the affected tests are still broken compared
to legacy, due to other reasons, e.g. missing <image> support).

* Source/WebCore/rendering/svg/RenderSVGContainer.cpp:
(WebCore::RenderSVGContainer::layout):
(WebCore::RenderSVGContainer::calculateViewport): Deleted.
* Source/WebCore/rendering/svg/RenderSVGContainer.h:
(WebCore::RenderSVGContainer::isLayoutSizeChanged const):
(WebCore::RenderSVGContainer::didTransformToRootUpdate const):
(WebCore::RenderSVGContainer::updateLayoutSizeIfNeeded):
* Source/WebCore/rendering/svg/RenderSVGTransformableContainer.cpp:
(WebCore::RenderSVGTransformableContainer::graphicsElement const):
(WebCore::RenderSVGTransformableContainer::additionalContainerTranslation const):
(WebCore::RenderSVGTransformableContainer::updateFromStyle):
(WebCore::RenderSVGTransformableContainer::updateLayerTransform):
(WebCore::RenderSVGTransformableContainer::applyTransform const):
(WebCore::RenderSVGTransformableContainer::calculateViewport): Deleted.
(WebCore::RenderSVGTransformableContainer::layoutChildren): Deleted.
(WebCore::RenderSVGTransformableContainer::styleWillChange): Deleted.
* Source/WebCore/rendering/svg/RenderSVGTransformableContainer.h:

Canonical link: https://commits.webkit.org/252466@main
